### PR TITLE
feat: support matching stdout for system

### PR DESCRIPTION
**Pull request by xxchan**

### CODE REVIEW

File: sqllogictest/src/runner.rs [hunk 1]
```diff
@@ -611,19 +642,33 @@ impl<D: AsyncDB, M: MakeConnection<Conn = D>> Runner<D, M> {
...
                 };
+                cmd.stdout(std::process::Stdio::piped());
+                cmd.stderr(std::process::Stdio::piped());
                 let result = D::run_command(cmd).await;
-
                 #[derive(thiserror::Error, Debug)]
-                #[error("process exited unsuccessfully: {0}")] // message from unstable `ExitStatusError`
-                struct SystemError(ExitStatus);
+                #[error("process exited unsuccessfully: {0}\nstdout: {1}\nstderr: {2}")]
+                struct SystemError(ExitStatus, String, String);
```
Comment (BugenZhao) [inline sqllogictest/src/runner.rs:650]:
```suggestion
                struct SystemError { 
	                status: ExitStatus,
	                stdout: String, 
	                stderr: String
                };
```

File: tests/system_command/system_command.slt [hunk 2]
```diff
@@ -15,3 +15,32 @@ query T
...
+
+# Note: it ends with 2 newlines
+system ok
+echo "114514"
+----
+114514
+
+
+# Note: when substitution is on, we need to escape the backslash
+# Note: 1 blank line in the middle is ok, but not 2
```
Comment (BugenZhao) [inline tests/system_command/system_command.slt:27]: What if the system command really contain 2 consecutive blank lines? 😕

**Review by wangrunji0408 — APPROVED** [review]

**Review by BugenZhao — APPROVED** [review]
Good

File: tests/system_command/system_command.slt (hunk 2 above)
Comment (xxchan) [reply tests/system_command/system_command.slt:27]: ntfs

**Pull request merged**